### PR TITLE
{lib}[fosscuda/2019b] TensorFlow v2.1.0 + pybind11 v2.4.3 w/ Python 3.7.4

### PR DESCRIPTION
**Pull request by Flamefire**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by Flamefire** [comment]
@boegel TF 2.1 needs a newer scipy, TF 2.0 is barely usable due to some bugs in the distributed implementation. How to proceed?

**Comment by boegel** [comment]
@Flamefire It looks like the issue could be specific to `scipy` 1.4.0, see https://github.com/tensorflow/tensorflow/commit/54daf3c5700897a6062313983933ca28e92c410d and in particular https://github.com/scipy/scipy/issues/11237...

One option could be to include a newer `scipy` in `exts_list` (with some comments to explain why).

**Comment by Flamefire** [comment]
Sounds reasonable. 👍

**Comment by boegel** [comment]
@Flamefire Did you actually run into problems when just using `SciPy-bundle` 2019.10 (which includes `scipy` 1.3.1?

Based on the various bug reports, it seems like the issue only occurs with `scipy` 1.4.0.

Also, https://github.com/pybind/pybind11/issues/1262 tells me it's only actually an issue when stuff is built with different compilers.

**Comment by Flamefire** [comment]
Yes: TF 2.1 requires the new SciPy which makes `pip check` fail

**Comment by boegel** [comment]
@Flamefire OK, but the `pip check` fail is only because of the hardcoded `scipy==1.4.1` used in `tensorflow/tools/ci_build/builds/pip_new.sh` (which isn't 100% correct imho).

But fine, I guess we can include `scipy` 1.4.1 in `exts_list` in the TensorFlow easyconfig as an escape hatch...

**Comment by Flamefire** [comment]
Sounds reasonable. Done

**Comment by Flamefire** [comment]
Test report by @Flamefire
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in this PR)
taurusa9 - Linux SL 7.7, Intel(R) Xeon(R) CPU E5-2603 v4 @ 1.70GHz, Python 2.7.5
See https://gist.github.com/3c17c1fef7a8067c99635957dd7b5fac for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
taurusml2 - Linux RHEL 7.6, 8335-GTX, Python 2.7.5
See https://gist.github.com/949af9f56203daf9951f36d38429f45f for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 2 out of 3 (2 easyconfigs in this PR)
node3170.skitty.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/34f3185c07e1736e0767f92b1d61a3ce for a full test report.

File: easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.1.0-fosscuda-2019b-Python-3.7.4.eb [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+name = 'TensorFlow'
+version = '2.1.0'
+versionsuffix = '-Python-%(pyver)s'
+
+homepage = 'https://www.tensorflow.org/'
+description = "An open-source software library for Machine Intelligence"
+
+toolchain = {'name': 'fosscuda', 'version': '2019b'}
+toolchainopts = {'usempi': True}
```
Comment (boegel) [inline easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.1.0-fosscuda-2019b-Python-3.7.4.eb]: @Flamefire To fix the `scipy` issue on x86_64, you'll need to enable `pic` here too (like we do in `SciPy-bundle`):

```python
toolchainopts = {'pic': True, 'usempi': True}
```

Comment (Flamefire) [reply easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.1.0-fosscuda-2019b-Python-3.7.4.eb]: Done and tested during the night.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in this PR)
node3107.skitty.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/70a8eb84040362b843e8cb1f64124f03 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in this PR)
node2509.golett.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/691915c87ee9126bfe24bb3686370052 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 3 out of 4 (2 easyconfigs in this PR)
node3304.joltik.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6242 CPU @ 2.80GHz, Python 3.6.8
See https://gist.github.com/6cb4df492f1db3fd634e5fa7b9819fbe for a full test report.

File: easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.1.0-fosscuda-2019b-Python-3.7.4.eb [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    }),
+    ('astor', '0.8.0', {
+        'checksums': ['37a6eed8b371f1228db08234ed7f6cfdc7817a3ed3824797e20cbb11dc2a7862'],
+    }),
+    ('Keras-Applications', '1.0.8', {
+        'modulename': 'keras_applications',
+        'source_tmpl': 'Keras_Applications-%(version)s.tar.gz',
+        'checksums': ['5579f9a12bcde9748f4a12233925a59b93b73ae6947409ff34aa2ba258189fe5'],
+    }),
+    ('gast', '0.2.2', {
```
Comment (boegel) [inline easybuild/easyconfigs/t/TensorFlow/TensorFlow-2.1.0-fosscuda-2019b-Python-3.7.4.eb]: @Flamefire We had a comment above here, please add it back:

```
# Tensorflow needs 0.2.2, see https://github.com/tensorflow/tensorflow/issues/32319
```

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node3304.joltik.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6242 CPU @ 2.80GHz, Python 3.6.8
See https://gist.github.com/f906a58822c46b6705d9e3b37b3eb42f for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2509.golett.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/e12482637802ecd1bb884064d9a1f3b2 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node3107.skitty.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/01511fc8a9f94138f115f771c4ac5390 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**FAILED**
Build succeeded for 0 out of 2 (2 easyconfigs in this PR)
taurusml2 - Linux RHEL 7.6, 8335-GTX, Python 2.7.5
See https://gist.github.com/688fc4bafd6929a9516c9db6855897d0 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
taurusml2 - Linux RHEL 7.6, 8335-GTX, Python 2.7.5
See https://gist.github.com/c18629856dd6f2d60f5073b9b77d6780 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @Flamefire!

**Comment by boegel** [comment]
@migueldiascosta FYI: squeezing this in for EasyBuild v4.1.1 (`4.1.x` is updated too)

**Comment by Flamefire** [comment]
Sorry that it took so long. Building with nproc=176 doesn't work and stalls the build. Had to reduce it to like 60. Not sure if EB could handle this automatically or if one is supposed to use max-parallel (or so)?

**Comment by boegel** [comment]
@Flamefire We could specify `maxparallel` in the easyconfig file (or even through the easyblock)

**Comment by Flamefire** [comment]
Ok, I opened #1934 for that.

**Pull request merged**